Formula/cog.rb

```diff
@@ -0,0 +1,35 @@
+class Cog < Formula
+  desc "Containers for machine learning"
+  homepage "https://github.com/replicate/cog"
+  url "https://github.com/replicate/cog.git",
+      tag:      "v0.5.1",
+      revision: "b58bd63e3cd9b7db0a3410ef47c61c1b11255809"
+  license "Apache-2.0"
+
+  head "https://github.com/replicate/cog.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+  depends_on "python@3.10" => :build
+  depends_on "redis"
+
+  def install
+    args = %W[
+      COG_VERSION=#{version}
+      PYTHON=python3
+    ]
+
+    system "make", *args
+    bin.install "cog"
+
+    generate_completions_from_executable(bin/"cog", "completion")
+  end
+
+  test do
+    assert_match "cog version #{version}", shell_output("#{bin}/cog --version")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
We can try a command with missing dependencies and check that we get the expected error code/message.

Reply:
We pass `COG_VERSION` to make, so let's keep the `--version` check (to make sure that the `test` block catches upstream build system changes in specifying the version). But doing `cog build` as well is good.

Unless it exits with `0` you'll need to pass a second argument to `shell_output` specifying the exit code to expect.